src/openms/include/OpenMS/ANALYSIS/OPENSWATH/OpenSwathOSWWriter.h

```diff
@@ -125,13 +125,14 @@ namespace OpenMS
   public:
 
     OpenSwathOSWWriter(const String& output_filename,
+                       const UInt64 run_id,
                        const String& input_filename = "inputfile",
                        bool ms1_scores = false,
                        bool sonar = false,
                        bool uis_scores = false) :
       output_filename_(output_filename),
       input_filename_(input_filename),
-      run_id_(OpenMS::UniqueIdGenerator::getUniqueId()),
+      run_id_(static_cast<int64_t>(run_id & ~(1ULL << 63))),  // conversion from UInt64 to int64_t to support SQLite (and conversion to 63 bits)
```

Comment:
I think this conversion is used at a few places in sqlite code. can we factor that out?

Reply:
there were only 3 places left, but I refactored into a single function.
While doing that, I researched a bit on WHY this is actually necessary, and the short answer is: because we're doing it wrong.
We INSERT into a db by creating a large string which contains all the data. This is slow and forces us to remove the 'sign'-bit of a 64bit unsigned int because Sqlite cannot store a large 64bit uint into its signed64bit int (it does not have uint64). Thus, it will fall back to store it as double(!), which of course in accurate for large integers.
To do it right we'd need to `bind` variables to prepared statements. This would not only be faster (since we safe the roundtrip via strings), but also allows to store the 64bit integer - it will just show up as negative value in Sqlite, but can be restored without loss if information when reading it back... but to fix that would take days.

Reply:
thanks for looking into that...  maybe interesting for @hroest